File: backend/versuion/src/main/java/com/example/versuion/models/LigneVente.java
package com.example.versuion.models;

import lombok.*;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import java.math.BigDecimal;

@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
@Entity
public class LigneVente extends AbstractEntity{

    @ManyToOne
    @JoinColumn(name = "idVent")
    private Ventes vente;

    private BigDecimal quantie;

    @ManyToOne
    @JoinColumn(name = "idarticle")
    private Article article;

    private BigDecimal prixUnitare;

    @Column(name = "identreprise")
    private Integer idEntreprise;
}

File: frontend/my-first-project/src/app/Models/LigneVente.ts
import { Article } from "./Article";
import { Ventes } from "./Ventes";

export interface LigneVente {
  id?: number;
  creationDate?: number;
  lastModifiedDate?: number;
  vente?: Ventes;
  article?: Article;
  quantite?: number;
  prixUnitaire?: number;
  idEntreprise?: number;
}
